### PR TITLE
CavcRemand add column source_appeal

### PR DESCRIPTION
**Pull request by lomky**
Bumps #15719 / [CASEFLOW-688](https://vajira.max.gov/browse/CASEFLOW-688)

### Description
Adds the new column `source_appeal`
Dupes code that refers to `appeal` to refer to `source_appeal`

All PRs:
CavcRemand add column source_appeal - This PR
Removes appeal_id from cavc_remand - #15777 
Adds cavc_appeal_id column - TK

### Acceptance Criteria
- [ ] Tests Pass

### Database Changes
*Only for Schema Changes*

* [x] Column comments updated
* [x] Have your migration classes inherit from `Caseflow::Migration`, especially when adding indexes (use `add_safe_index`)
* [x] Verify that `migrate:rollback` works as desired ([`change` supported functions](https://edgeguides.rubyonrails.org/active_record_migrations.html#using-the-change-method))
* [x] Appropriate indexes added (especially for foreign keys, polymorphic columns, unique constraints, and Rails scopes)
* [x] DB schema docs updated with `make docs` (after running `make migrate`)
* [x] #appeals-schema notified with summary and link to this PR

### CODE REVIEW

File: spec/factories/cavc_remand.rb [hunk 1]
```diff
@@ -15,13 +15,14 @@
     after(:build) do |cavc_remand, evaluator|
       cavc_remand.created_by = (evaluator.created_by || User.first)
 
-      if evaluator.appeal
-        cavc_remand.appeal = evaluator.appeal
+      if evaluator.source_appeal
+        cavc_remand.source_appeal = evaluator.source_appeal
+        cavc_remand.appeal = evaluator.source_appeal
```
Comment (yoomlam) [inline spec/factories/cavc_remand.rb:20]: I'm guessing `cavc_remand.appeal` will be renamed or removed.

**Review by yoomlam — DISMISSED** [review]
LGTM!

File: spec/factories/cavc_remand.rb (hunk 1 above)
Comment (lomky) [reply spec/factories/cavc_remand.rb:20]: Yup!

**Review by yoomlam — APPROVED** [review]
Still looks good!

**Pull request merged**